### ISSUE TITLE
Performance of model

### ISSUE THREAD

**Issue by donghyeops** [yikang-li/MSDN#5]
Hi Yikang.
I am studying your code.

I ran training code from provided RPN model.
and evaluate trained model using 'eval.sh'

but, I got low performance than provided full model.
My trained model's performance is as follows.
Recall@50 : 8.772%
Recall@100 : 10.908%

Used training command is 
CUDA_VISIBLE_DEVICES=0 python train_hdn.py \
    --load_RPN \
    --saved_model_path=./output/RPN/RPN_region_full_best.h5  \
    --dataset_option=normal --enable_clip_gradient \
    --step_size=2 \
    --MPS_iter=1 \
    --caption_use_bias \
    --caption_use_dropout \
    --rnn_type LSTM_normal 

How can I get normal performance?

Also I found difference from between git code and paper.
paper's MPS_iter is 2, but code's baseline is 1.
paper's message passing method is Message_Passing_Unit_v2(add) , but code's baseline is Message_Passing_Unit_v1(mean).
Could you let me know why there are different?

**Comment by yikang-li** [comment]
Hi, 

1. For the performance:
There is a trick here: you should extend the training epochs from the checkpoint using [--resume_training] (similar operations are done on the baseline methods). You could have a try and let me know if it could not produce the similar results. 

2. About the message passing unit:
For a fixed gate width (G in the paper), [sum] and [mean] perform similarly. We ignore the denominator in the paper to save space. 

Moreover, MP_unit_v2 is not as you understand. The [Sum] here is not to sum up along the "gate" dim, but for the dot product. We use the dot product of the two transformed features to determine whether the gate is open or not. After we apply the element-wise product, we should add them up (yes, that is the [sum] you said). We call it [kernel method], which is one of our attempts but not shown in the paper. I apologize we forgot to clean that part and bring you some confusion. 

Best,
Yikang

**Comment by donghyeops** [comment]
Thanks a lot for your reply.
